libqtile/xcbq.py

```diff
@@ -316,20 +317,20 @@ def alloc_color(self, color):
         """
             Flexible color allocation.
         """
-        if color.startswith("#"):
-            if len(color) != 7:
+        try:
+            return self.conn.conn.core.AllocNamedColor(
+                self.cid, len(color), color
+            ).reply()
+        except xcffib.xproto.NameError:
+            if len(color) < 6:
                 raise ValueError("Invalid color: %s" % color)
```

Comment:
I wonder if we really need this test... there can be more things wrong in the `color` string than its length, and IMHO the built-in exception that would be normally raised in those cases would be enough.